sources/file.go

```diff
@@ -0,0 +1,247 @@
+package sources
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/h2non/filetype"
+	"github.com/mholt/archives"
+
+	"github.com/zricethezav/gitleaks/v8/config"
+	"github.com/zricethezav/gitleaks/v8/logging"
+)
+
+const defaultBufferSize = 100 * 1_000 // 100kb
+const InnerPathSeparator = "!"
+
+type seekReaderAt interface {
+	io.ReaderAt
+	io.Seeker
+}
+
+// File is a source for yielding fragments from a file or other reader
+type File struct {
+	// Content provides a reader to the file's content
+	Content io.Reader
+	// Path is the resolved real path of the file
+	Path string
+	// Symlink represents a symlink to the file if that's how it was discovered
+	Symlink string
+	// Buffer is used for reading the content in chunks
+	Buffer []byte
+	// Config is the gitleaks config used for shouldSkipPath. If not set, then
+	// shouldSkipPath is ignored
+	Config *config.Config
+	// outerPaths is the list of container paths (e.g. archives) that lead to
+	// this file
+	outerPaths []string
+	// MaxArchiveDepth limits how deep the sources will explore nested archives
+	MaxArchiveDepth int
+	// archiveDepth is the current archive nesting depth
+	archiveDepth int
+}
+
+// Fragments yields fragments for the this source
+func (s *File) Fragments(ctx context.Context, yield FragmentsFunc) error {
+	format, _, err := archives.Identify(ctx, s.Path, nil)
+	if err == nil && format != nil {
```

Comment:
If `err != nil` we swallow it. Is that intended? If so, probably worth a comment.

Reply:
Yeah it was intended. Adding a comment :ok_hand: